From bb5ccd68080a40672c0820783c2f3334a950b7d5 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Thu, 29 Dec 2016 00:22:57 +0700
Subject: [PATCH 78/81] Revert "frameworks/base: Don't report existing menu key
 if it was remapped. [2/3]"

This reverts commit eb5c5510196874bfc05c730bd1fc30c888f515ce.

Change-Id: Ib50b2c38856f8a5658f135873380e7cbb96eface
---
 core/java/android/view/IWindowManager.aidl                        | 5 -----
 core/java/android/view/ViewConfiguration.java                     | 2 +-
 core/java/android/view/WindowManagerPolicy.java                   | 1 -
 .../core/java/com/android/server/policy/PhoneWindowManager.java   | 8 --------
 .../core/java/com/android/server/wm/WindowManagerService.java     | 5 -----
 5 files changed, 1 insertion(+), 20 deletions(-)

diff --git a/core/java/android/view/IWindowManager.aidl b/core/java/android/view/IWindowManager.aidl
index 0c39318..d3eaef3 100644
--- a/core/java/android/view/IWindowManager.aidl
+++ b/core/java/android/view/IWindowManager.aidl
@@ -352,11 +352,6 @@ interface IWindowManager
     boolean hasNavigationBar();
 
     /**
-     * Simulate a hardware menu key
-     */
-    boolean hasPermanentMenuKey();
-
-    /**
      * Lock the device immediately with the specified options (can be null).
      */
     void lockNow(in Bundle options);
diff --git a/core/java/android/view/ViewConfiguration.java b/core/java/android/view/ViewConfiguration.java
index b8a803b..3ea547b 100644
--- a/core/java/android/view/ViewConfiguration.java
+++ b/core/java/android/view/ViewConfiguration.java
@@ -373,7 +373,7 @@ public class ViewConfiguration {
                 case HAS_PERMANENT_MENU_KEY_AUTODETECT: {
                     IWindowManager wm = WindowManagerGlobal.getWindowManagerService();
                     try {
-                        sHasPermanentMenuKey = wm.hasPermanentMenuKey();
+                        sHasPermanentMenuKey = !wm.hasNavigationBar();
                         sHasPermanentMenuKeySet = true;
                     } catch (RemoteException ex) {
                         sHasPermanentMenuKey = false;
diff --git a/core/java/android/view/WindowManagerPolicy.java b/core/java/android/view/WindowManagerPolicy.java
index 1dc5165..b4a7211 100644
--- a/core/java/android/view/WindowManagerPolicy.java
+++ b/core/java/android/view/WindowManagerPolicy.java
@@ -1321,7 +1321,6 @@ public interface WindowManagerPolicy {
      * Specifies whether there is an on-screen navigation bar separate from the status bar.
      */
     public boolean hasNavigationBar();
-    public boolean hasPermanentMenuKey();
 
     /**
      * Device requires a software navigation bar.
diff --git a/services/core/java/com/android/server/policy/PhoneWindowManager.java b/services/core/java/com/android/server/policy/PhoneWindowManager.java
index f4093fa..e52d35d 100644
--- a/services/core/java/com/android/server/policy/PhoneWindowManager.java
+++ b/services/core/java/com/android/server/policy/PhoneWindowManager.java
@@ -890,7 +890,6 @@ public class PhoneWindowManager implements WindowManagerPolicy {
 
     private boolean mVolumeAnswerCall;
 
-    private boolean mHasPermanentMenuKey;
     private class PolicyHandler extends Handler {
         @Override
         public void handleMessage(Message msg) {
@@ -9145,19 +9144,12 @@ public class PhoneWindowManager implements WindowManagerPolicy {
         return mNavbarVisible;
     }
 
-    // reflects factory or emulator prop override
-    // basically a cm compatability shim
     @Override
     public boolean needsNavigationBar() {
         return mHasNavigationBar;
     }
 
     @Override
-    public boolean hasPermanentMenuKey() {
-        return !hasNavigationBar() && mHasPermanentMenuKey;
-    }
-
-    @Override
     public void setLastInputMethodWindowLw(WindowState ime, WindowState target) {
         mLastInputMethodWindow = ime;
         mLastInputMethodTargetWindow = target;
diff --git a/services/core/java/com/android/server/wm/WindowManagerService.java b/services/core/java/com/android/server/wm/WindowManagerService.java
index 0ebf0fc..7da3ed8 100644
--- a/services/core/java/com/android/server/wm/WindowManagerService.java
+++ b/services/core/java/com/android/server/wm/WindowManagerService.java
@@ -10470,11 +10470,6 @@ public class WindowManagerService extends IWindowManager.Stub
     }
 
     @Override
-    public boolean hasPermanentMenuKey() {
-        return mPolicy.hasPermanentMenuKey();
-    }
-
-    @Override
     public void lockNow(Bundle options) {
         mPolicy.lockNow(options);
     }
-- 
2.9.3

